src/OrchardCore/OrchardCore.Abstractions/Base64.cs

```diff
@@ -51,7 +51,7 @@ public static Stream DecodedToStream(string base64)
         }
 
         memoryStream.Advance(bytesWritten);
-
+        memoryStream.Seek(0, SeekOrigin.Begin);
```

Comment:
```suggestion
        memoryStream.Seek(0, SeekOrigin.Begin);
        
```

Reply:
Isn't this the same are just removing the `memoryStream.Advance(bytesWritten);`?

Reply:
No, `Advance` is the pattern used for `IBufferWriter`. This updates the stream with how many bytes were written to the span.

Reply:
Just for anyone wondering how it works.

`MemoryStreamFactory.GetStream(maxBufferLength)` returns a stream with a pooled backing `byte[]` of size `maxBufferLength`. The cursor is at position `0`, the length of the stream is `0`, since nothing was written to it. But there are `maxBufferLength` for now to write into.

`memoryStream.GetSpan(maxBufferLength)` return a `Span<byte>` which is like a pointer to the memory we can write to (the memory of the pooled backing `byte[]`), and it's limited to `maxBufferLength` bytes. We can't get a bigger span at that point (obviously).

`Convert.TryFromBase64String(base64, span, out var bytesWritten)` writes `bytesWritten` to the Span. It won't be more that the limit we know.

Now we need to tell the stream that we wrote `bytesWritten` bytes into the Span, so it should update its internal cursor such that the next call to `Write` or `GetSpan` should return a pointer at that new location, and not overwrite what we just did. To say that we call `Advance(bytesWritten)`.

Now we are done with writing, we return the stream pointing at the beginning. The receiver should not write to it, just read it.